src/runtime-support.cc

```diff
@@ -312,32 +312,70 @@ inline rt::cookie::FileState* _file_state(rt::Cookie* cookie) {
         throw rt::ValueUnavailable("no current connection or file available");
 }
 
-static std::string _file_id(rt::Cookie* cookie) {
-    assert(cookie);
-    return _file_state(cookie)->id();
+inline rt::cookie::FileState* _file_state(rt::Cookie* cookie, std::optional<std::string> fid) {
+    auto* stack = _file_state_stack(cookie);
+    if ( fid ) {
+        if ( auto* fstate = stack->find(*fid) )
+            return fstate;
+        else
+            throw rt::ValueUnavailable(hilti::rt::fmt("no file analysis currently in flight for file ID %s", fid));
+    }
+    else {
+        if ( stack->isEmpty() )
+            throw rt::ValueUnavailable("no file analysis currently in flight");
+
+        return stack->current();
+    }
+}
+
+rt::cookie::FileState* rt::cookie::FileStateStack::push() {
+    auto fid = ::zeek::file_mgr->HashHandle(hilti::rt::fmt("%s.%d", _analyzer_id, ++_id_counter));
```

Comment:
Can we instead of maintaining an `_id_counter` add a member function `size_t size() const` instead which just exposes `_stack`'s size and use it here? That would make the naming 0-based, but nothing else should change.

Reply:
No, we need to keep generating globally unique IDs for new analyzers, independent of stack size.